Sharding strategy and recommendations (#355)
crate/cratedb-guide: docs/admin/sharding-partitioning.md

We can see that there are now 8 shards for the table `second_table` in the
cluster.

:::{danger}
**Over-sharding and over-partitioning**

Sharding can drastically improve the performance on large datasets.
However, having too many small shards will most likely degrade performance.
Over-sharding and over-partitioning are common flaws leading to an overall
poor performance.

**As a rule of thumb, a single shard should hold somewhere between 5 - 50
GB of data.**

To avoid oversharding, CrateDB by default limits the number of shards per
node to 1000. Any operation that would exceed that limit, leads to an
exception.

> :::

Reviewer: Why remove this? Isn't it good to have as a top highlighted note? Also it correctly mentions `by default ... 1000 shards` while the text below is missing this `default` part.

- Identify the record size
- Calculate the throughput

Then, to calculate the number of shards, consider that each shard should
roughly be between 10 – 50 GB, and that each node can manage
up to 1_000 shards by default.

:::{caution}
**Over-sharding and over-partitioning**

Sharding can drastically improve the performance on large datasets.
However, having too many small shards will most likely degrade performance.
Over-sharding and over-partitioning are common flaws leading to an overall
poor performance.

Learn how to discover an optimal sharding strategy for your dataset
in the {ref}`sharding-guide`.
:::

## Example

> Let's create a basic sharding strategy on behalf of a concrete example. Imagine

Reviewer 2: ```suggestion
Let's create a basic sharding strategy by using a concrete example. Imagine
```
I think it reads better like that.

Author: Thanks, fixed with d4ca40a66.

- Calculate the throughput

Then, to calculate the number of shards, you should consider that the size of each
shard should roughly be between 10 - 50 GB, and that each node can only manage

> up to 1000 shards.

Author: Improved per 95dba4d12ef.
